specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json

```diff
@@ -1446,6 +1633,74 @@
       },
       "description": "A list of firewall rules."
     },
+    "VirtualNetworkRuleProperties": {
+      "description": "Properties of a virtual network rule.",
+      "required": [
+        "virtualNetworkSubnetId"
+      ],
+      "type": "object",
+      "properties": {
+        "virtualNetworkSubnetId": {
+          "description": "The ARM resource id of the virtual network subnet.",
+          "type": "string"
+        },
+        "ignoreMissingVnetServiceEndpoint": {
+          "description": "Create firewall rule before the virtual network has vnet service endpoint enabled.",
+          "type": "boolean"
+        },
+        "state": {
```

Comment:
This should be replaced by the standard "provisioningState" property that ARM recommends to use to indicate the state of the resource during creation.

Reply:
for now we are keeping this in sync with Microsoft.SQL Resource Provider for consistency .

Reply:
Consistency with 100+ providers is what we should be trying to achieve, not with one :). Customers would definitely love that more. Not blocking on it. But please note about such things as it impacts the broader Azure REST API usability, consistency and client experience.

Reply:
Thanks , I will look into this soon.